tests/unit/swagger/definitionsGeneration/metadata.spec.ts

```diff
@@ -474,6 +476,27 @@ describe('Metadata generation', () => {
       expect(parameter.parameterName).to.equal('firstname');
       expect(parameter.required).to.be.true;
     });
+
+    it('Should inline enums for TS Enums in path, query and header when using Swagger', () => {
```

Comment:
Does this test still pass without the refactoring that you did? I'm trying to make sense of if this is a change in behavior or a functionality change. I don't care which one it is, I'm just trying to understand so I can better serve the community if we get questions later.

Reply:
Well there are 2 things, I shoulda explained earlier.
While the end result should be equal in terms of swagger docs, the intermediate representation does change (the metadata).

I'm trying to defer the special case extractEnum tried to solve from the resolver that builds the metadata to the specGenerator for the Swagger2 specification that takes the metadata and produces the swagger.json.

Reply:
That means, all the tests that check the Swagger output (and OpenAPI aswell) stay unchanged.
However, I had to adapt the tests that perform assertions on the metadata.